sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/CustomEventLoggerExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Tests.CommonTestFramework
+{
+#if !NET6_0 // The TagName class isn't available in NET6.
+
+    // TODO: REMOVE THIS PRAGMA AFTER Microsoft.Extensions.Telemetry SHIPS THE TagName API AS STABLE.
+    // https://github.com/dotnet/extensions/issues/5825
+#pragma warning disable EXTEXP0003 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+    /// <summary>
+    /// This class has ILogger extensions for use in unit tests.
+    /// </summary>
+    public static partial class CustomEventLoggerExtensions
+    {
+        [LoggerMessage(level: LogLevel.Information, Message = "{key1}")]
+        public static partial void WriteSimpleLog(this ILogger logger, string key1);
```

Comment:
check if you can borrow the idea from here https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/logs/complex-objects. This does not require additional libraries.

Reply:
I don't believe this will work. It's the same problem... we need to have a specific attribute key which includes special characters.